kube-runtime/src/controller/mod.rs

```diff
@@ -1085,6 +1089,89 @@ where
         self
     }
 
+    /// Trigger a reconciliation for all managed objects having a user defined relation to `Other`.
+    /// Mapper method accepts both managed and `Other` objects, allowing to determine if objects are related.
+    ///
+    /// For example, this can be used to reconcile all managed objects which have an annotation with name of the `Other` object.
+    /// Usefull when there is a chain of ownership references, where the parent should trigger reconcile for all of the most distant children:
+    ///
+    /// ```
+    /// # async {
+    /// use futures::stream::StreamExt;
+    /// use k8s_openapi::api::core::v1::ConfigMap;
+    /// use k8s_openapi::api::apps::v1::Deployment;
+    /// use kube::{
+    ///     Client,
+    ///     api::{Api, ResourceExt},
+    ///     runtime::{
+    ///         reflector::ObjectRef,
+    ///         controller::{Controller, Action},
+    ///         watcher,
+    ///     },
+    /// };
+    /// use std::{convert::Infallible, io::BufRead, sync::Arc};
+    /// Controller::new(
+    ///     Api::<ConfigMap>::all(Client::try_default().await.unwrap()),
+    ///     watcher::Config::default(),
+    /// )
+    /// .watches_all(
+    ///     Api::<Deployment>::all(Client::try_default().await.unwrap()),
+    ///     watcher::Config::default(),
+    ///     |config_map, deployment| match deployment.annotations().get("config") {
+    ///         Some(name) if config_map.name_any() == name.clone() => Some(ObjectRef::from_obj(config_map)),
+    ///         _ => None,
+    ///     },
```

Comment:
This looks like a [`Controller::watches_stream`](https://docs.rs/kube/latest/kube/runtime/struct.Controller.html#method.watches_stream), but it also seems to bake in a type of __filter__ against a the root controller object (configmap)?